autorest/codegen/__init__.py

```diff
@@ -86,7 +86,8 @@ def _create_code_model(self, yaml_data: Dict[str, Any], options: Dict[str, Union
         else:
             dollar_host_parameter = dollar_host[0]
             code_model.global_parameters.remove(dollar_host_parameter)
-            code_model.base_url = dollar_host_parameter.yaml_data["clientDefaultValue"]
+            code_model.base_url = True
```

Comment:
repurpose code model's `base_url` property to signify if there's a base url, and add property `base_url_default_value` to actually hold the default value